### ISSUE TITLE
Use ES6 throughout

### ISSUE THREAD

**Issue by souporserious** [HaikuTeam/core#1]
First, congrats on open sourcing this! Really excited to see where this project goes. I was curious if there is a roadmap for converting things to use ES6? I'd love to take a stab at converting things if possible, seems simple enough and I'd love to learn more about the interworkings of everything 😇please let me know and I can start working on a PR.

**Comment by matthewtoast** [comment]
Hi @souporserious - thanks so much for your interest!

As it happens, we are already in progress with that task ourselves. (And... I made a mistake when I wrote the roadmap list: We're actually going to _stick with TypeScript_, but we are working on reshaping it to use more correct "ES6-flavored" constructs like `class` and `=>`.)

Shortly I'm going to make it clear in the README which tasks we've already got baking.

But...We're stoked by your offer to contribute. Let me know if there's another aspect of this project you're interested in, or if I can give any pointers about this codebase. 👍 Hoping to expand the docs soon so it's more clear how the pieces fit together.
